cassiopeia/solver/SpectralNeighborJoiningSolver.py

```diff
@@ -0,0 +1,281 @@
+"""
+This file stores a subclass of DistanceSolver, SpectralNeighborJoining. This algorithm is based on the one developed by Jaffe, et al. (2020) in their paper titled "Spectral neighbor joining for reconstruction of latent tree models. 
+"""
+
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import abc
+import networkx as nx
+import numpy as np
+import pandas as pd
+from itertools import chain
+
+# Only for debugging purposes
+# if True:
+#     import sys
+#     sys.path.insert(0, '../..')
+
+from cassiopeia.solver.DistanceSolver import DistanceSolver, DistanceSolverError
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.solver import (
+    dissimilarity_functions,
+    solver_utilities,
+)
+
+from sklearn.decomposition import TruncatedSVD
+from scipy.sparse.linalg import svds, eigs
+from scipy.linalg import svd as scipy_svd
+
+
+class SpectralNeighborJoiningSolver(DistanceSolver):
+    """
+    Spectral Neighbor-joining class for Cassiopeia.
+
+    Implements a variation on the Spectral Neighbor-Joining algorithm described by Jaffe et al. in their 2020 paper. This class implements the abstract class DistanceSolver. This class inherits the 'solve' method from DistanceSolver with the find_cherry method implemented using a spectral method as in Jaffe et al. (2020).
+
+    Args:
+        dissimilarity_function: A function by which to compute the dissimilarity map, corresponding to the affinity R(i, j) in Jaffe et al. (2020). By default we will use exp(-d(i,j)) where d is the metric given by weighted_hamming_distance from dissimilarity functions without weights given.
+    """
+
+    def __init__(
+        self,
+        dissimilarity_function: Optional[
+            Callable[
+                [np.ndarray, np.ndarray, int, Dict[int, Dict[int, float]]], float
+            ]
+        ] = dissimilarity_functions.weighted_hamming_distance,
+        add_root: bool = False,
+        prior_transformation: str = "negative_log",
+    ):
+        super().__init__(dissimilarity_function, add_root=add_root, prior_transformation=prior_transformation) #type: ignore
+
+    def get_dissimilarity_map(
+        self, 
+        cassiopeia_tree: CassiopeiaTree,
+        layer: Optional[str] = None
+    ) -> pd.DataFrame: 
+        """Outputs the first lambda matrix, where every subset is an individual node.
+
+        Args:
+            cassiopeia_tree: the CassiopeiaTree object passed into solve()
+            layer: Layer storing the character matrix for solving. If None, the
+                default character matrix is used in the CassiopeiaTree.
+
+        Returns:
+            DataFrame object of the lamba matrix, but similar in structure to DistanceSolver's dissimilarity_map.
+        """
+
+        # get similarity map and save it as private instance variable
+        self.__similarity_map = super().get_dissimilarity_map(cassiopeia_tree, layer)
```

Comment:
Why do you assume that the `dissimilarity_map` stored in the `CassiopeiaTree` is a similarity map and not a dissimilarity map? This is not clear from the documentation, and may account for some of the integration issues that @kevinan1 is observing?